### PR TITLE
dependencies: updating `containerservice` to `2023-09-02-preview`

### PR DESCRIPTION
**Pull request by ms-henglu**
I found that for cluster, networkProfile.networkPolicy returns "none" in `2023-09-02-preview` instead of null. But in azurerm provider, this field is computed, so it won't cause a plan-diff.

## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [ ] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [ ] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
all tests are now failing with:
```
------- Stdout: -------
=== RUN   TestAccInferenceCluster_completeCustomSSL
=== PAUSE TestAccInferenceCluster_completeCustomSSL
=== CONT  TestAccInferenceCluster_completeCustomSSL
    testcase.go:113: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_kubernetes_cluster.test will be updated in-place
          ~ resource "azurerm_kubernetes_cluster" "test" {
                id                                  = "/subscriptions/*******/resourceGroups/acctestRG-ml-240506174925305018/providers/Microsoft.ContainerService/managedClusters/acctestaks240506174925305018"
                name                                = "acctestaks240506174925305018"
                # (26 unchanged attributes hidden)
        
              ~ default_node_pool {
                    name                         = "default"
                  + node_taints                  = []
                  + tags                         = {}
                  + zones                        = []
                    # (21 unchanged attributes hidden)
        
                  - upgrade_settings {
                      - max_surge = "10%" -> null
                    }
                }
        
              ~ identity {
                  + identity_ids = []
                    # (3 unchanged attributes hidden)
                }
        
                # (2 unchanged blocks hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccInferenceCluster_completeCustomSSL (779.74s)
FAIL
```

**Comment by ms-henglu** [comment]
Hi @katbyte , thanks, but the failed tests are unrelated with this change.

Most of the same failed tests are fixed in https://github.com/hashicorp/terraform-provider-azurerm/pull/25200. I'll open another one to fix the rest of them: https://github.com/hashicorp/terraform-provider-azurerm/pull/25877

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -1631,6 +1623,35 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
...
+					},
+					"vertical_pod_autoscaler_update_mode": {
+						Type:       pluginsdk.TypeString,
+						Computed:   true,
+						Deprecated: "The AKS API has removed support for this field on 2023-07-02-preview and is no longer possible to export this value.",
+					},
+					"vertical_pod_autoscaler_controlled_values": {
+						Type:       pluginsdk.TypeString,
+						Computed:   true,
+						Deprecated: "The AKS API has removed support for this field on 2023-07-02-preview and is no longer possible to export this value.",
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_resource.go:1650]: Can we make sure to specify that deprecated properties will be removed in the next major version of the provider 
```suggestion
						Deprecated: "The AKS API has removed support for this field on 2023-07-02-preview and is no longer possible to export this value. This property will be removed in v4.0 of the AzureRM provider.",
					},
					"vertical_pod_autoscaler_controlled_values": {
						Type:       pluginsdk.TypeString,
						Computed:   true,
						Deprecated: "The AKS API has removed support for this field on 2023-07-02-preview and is no longer possible to export this value. This property will be removed in v4.0 of the AzureRM provider.",
```

**Review by stephybun — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
Hi @stephybun , thanks for the code review.

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @ms-henglu. We should convert the `none` value that's being returned from the API for networkPolicy to an empty string in case users are using that value in a condition. Once that's done this should be good to go.

**Comment by ms-henglu** [comment]
Hi @stephybun , thanks! I've added a commit to address the comment.

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu LGTM 👍

**Comment by RolfMoleman** [comment]
Why is this functionality in a minor release rather than a major? As this causes fundamental issues with automated dependency management tools such as renovate or dependabot

**Pull request merged**